pkg/stanza/fileconsumer/file.go

```diff
@@ -211,9 +216,8 @@ func (m *Manager) makeFingerprint(path string) (*fingerprint.Fingerprint, *os.Fi
 // makeReader take a file path, then creates reader,
 // discarding any that have a duplicate fingerprint to other files that have already
 // been read this polling interval
-func (m *Manager) makeReaders(paths []string) []*reader.Reader {
-	readers := make([]*reader.Reader, 0, len(paths))
-OUTER:
+func (m *Manager) makeReaders(paths []string) {
+	m.activeFiles.Clear()
```

Comment:
I still think we shouldn't have to clear a file set, maybe ever.

```suggestion
	m.activeFiles = fileset.New[*reader.Reader](m.maxBatchFiles / 2)
```